editor/scene/scene_tree_editor.cpp

```diff
@@ -926,19 +926,20 @@ void SceneTreeEditor::_update_tree(bool p_scroll_to_selected) {
 		return;
 	}
 
-	Node *scene_node = get_scene_node();
+	if (!update_when_invisible && !is_visible_in_tree()) {
+		return;
+	}
 
-	if (node_cache.current_scene_node != scene_node) {
+	Node *scene_node = get_scene_node();
+	const ObjectID scene_id = scene_node ? scene_node->get_instance_id() : ObjectID();
+	if (node_cache.current_scene_id != scene_id) {
```

Comment:
What difference does it make? The ObjectID is only used for comparing, so it should be the same as pointer.